otsclient/cmds.py

```diff
@@ -484,14 +485,22 @@ def git_extract_command(args):
 
     stamper = GitTreeTimestamper(commit.tree)
 
+    path = args.path
+
+    # Use a relative path when the user is not in the current git repository
+    # top-level working directory and the user also provided a relative path to
+    # a file.
+    if not os.path.exists(os.path.join(repo_base_path, path)):
```

Comment:
This will probably fail for cases where you have similarly named files in the root path of the git working directory (whether committed or not).